core/testsupport/src/TestSupport.cxx

```diff
@@ -76,6 +76,20 @@ static struct ForbidDiagnostics {
         return;
       }
 
+      if (level == kWarning
+          && strcmp(location, "RIoUring") == 0
+          && strcmp(msg, "io_uring is unexpectedly not available because:\nFailed to allocate memory for the smallest possible io_uring instance. 'memlock' memory has been exhausted for this user") == 0) {
```

Comment:
I don't know if we only want to check a substring of the message. This would make the code shorter and also makes future mistakes less likely where we change a part of the message.